homeassistant/components/lyric/climate.py

```diff
@@ -0,0 +1,280 @@
+from time import gmtime, strftime, time
+from typing import List, Optional
+
+from aiolyric.objects.device import LyricDevice
+from aiolyric.objects.location import LyricLocation
+import voluptuous as vol
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    ATTR_TARGET_TEMP_HIGH,
+    ATTR_TARGET_TEMP_LOW,
+    HVAC_MODE_COOL,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_HEAT_COOL,
+    HVAC_MODE_OFF,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import entity_platform
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from . import LyricDeviceEntity
+from .const import (
+    DOMAIN,
+    LYRIC_EXCEPTIONS,
+    PRESET_HOLD_UNTIL,
+    PRESET_NO_HOLD,
+    PRESET_PERMANENT_HOLD,
+    PRESET_TEMPORARY_HOLD,
+    PRESET_VACATION_HOLD,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+
+LYRIC_HVAC_MODE_OFF = "Off"
+LYRIC_HVAC_MODE_HEAT = "Heat"
+LYRIC_HVAC_MODE_COOL = "Cool"
+LYRIC_HVAC_MODE_HEAT_COOL = "Auto"
+
+LYRIC_HVAC_MODES = {
+    HVAC_MODE_OFF: LYRIC_HVAC_MODE_OFF,
+    HVAC_MODE_HEAT: LYRIC_HVAC_MODE_HEAT,
+    HVAC_MODE_COOL: LYRIC_HVAC_MODE_COOL,
+    HVAC_MODE_HEAT_COOL: LYRIC_HVAC_MODE_HEAT_COOL,
+}
+
+HVAC_MODES = {
+    LYRIC_HVAC_MODE_OFF: HVAC_MODE_OFF,
+    LYRIC_HVAC_MODE_HEAT: HVAC_MODE_HEAT,
+    LYRIC_HVAC_MODE_COOL: HVAC_MODE_COOL,
+    LYRIC_HVAC_MODE_HEAT_COOL: HVAC_MODE_HEAT_COOL,
+}
+
+SERVICE_HOLD_TIME = "set_hold_time"
+ATTR_TIME_PERIOD = "time_period"
+
+SCHEMA_HOLD_TIME = {
+    vol.Required(ATTR_TIME_PERIOD, default="01:00:00"): vol.All(
+        cv.time_period,
+        cv.positive_timedelta,
+        lambda td: strftime("%H:%M:%S", gmtime(time() + td.total_seconds())),
+    )
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up the Honeywell Lyric climate platform based on a config entry."""
+    coordinator: DataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    entities = []
+
+    for location in coordinator.data.locations:
+        for device in location.devices:
+            entities.append(LyricClimate(hass, coordinator, location, device))
+
+    async_add_entities(entities, True)
+
+    platform = entity_platform.current_platform.get()
+
+    platform.async_register_entity_service(
+        SERVICE_HOLD_TIME,
+        SCHEMA_HOLD_TIME,
+        "async_set_hold_time",
+    )
+
+
+class LyricClimate(LyricDeviceEntity, ClimateEntity):
+    """Defines a Honeywell Lyric climate entity."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
```

Comment:
Don't pass in `hass`. It will be set on the entity when the entity has been added to home assistant. This means you can access `hass` with `self.hass`, after the entity has been added to home assistant.

Pass in the unit instead.